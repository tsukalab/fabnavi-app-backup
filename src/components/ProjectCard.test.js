// ProjectCard.test.js
import React from 'react';
import ProjectCard from './ProjectCard';
import renderer from 'react-test-renderer';

test('ProjectCard changes the class when hovered', () => {

    const project = {
        id: 100,
        name: 'sample',
        description: 'this is sample project',
        user: {
            image: 'https://avatars0.githubusercontent.com/u/12455918?v=4'
        }
    }
    const component = renderer.create(
        <ProjectCard {...project} toggleMenu={()=>{}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});