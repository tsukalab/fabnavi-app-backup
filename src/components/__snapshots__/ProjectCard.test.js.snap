// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard changes the class when hovered 1`] = `
<div
  data-jsx={679893488}
>
  <div
    className="project-box "
    data-jsx={679893488}
    onClick={undefined}
  >
    <div
      className="thumbnail"
      data-jsx={679893488}
    >
      <img
        data-jsx={679893488}
        src="./images/video-thumbnail.png"
      />
    </div>
    <h1
      className="project-name"
      data-jsx={679893488}
      title="sample"
    >
      sample
    </h1>
    <hr
      data-jsx={679893488}
    />
    <div
      className="box"
      data-jsx={679893488}
    >
      <img
        className="user-icon"
        data-jsx={679893488}
        src="https://avatars0.githubusercontent.com/u/12455918?v=4"
      />
      <div
        className="username"
        data-jsx={679893488}
      />
      <div
        className="date"
        data-jsx={679893488}
      >
        
      </div>
      <div
        className="description"
        data-jsx={679893488}
      >
        this is sample project
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectCard changes the class when hovered 2`] = `
<div
  data-jsx={679893488}
>
  <div
    className="project-box "
    data-jsx={679893488}
    onClick={undefined}
  >
    <div
      className="thumbnail"
      data-jsx={679893488}
    >
      <img
        data-jsx={679893488}
        src="./images/video-thumbnail.png"
      />
    </div>
    <h1
      className="project-name"
      data-jsx={679893488}
      title="sample"
    >
      sample
    </h1>
    <hr
      data-jsx={679893488}
    />
    <div
      className="box"
      data-jsx={679893488}
    >
      <img
        className="user-icon"
        data-jsx={679893488}
        src="https://avatars0.githubusercontent.com/u/12455918?v=4"
      />
      <div
        className="username"
        data-jsx={679893488}
      />
      <div
        className="date"
        data-jsx={679893488}
      >
        
      </div>
      <div
        className="description"
        data-jsx={679893488}
      >
        this is sample project
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectCard changes the class when hovered 3`] = `
<div
  data-jsx={679893488}
>
  <div
    className="project-box "
    data-jsx={679893488}
    onClick={undefined}
  >
    <div
      className="thumbnail"
      data-jsx={679893488}
    >
      <img
        data-jsx={679893488}
        src="./images/video-thumbnail.png"
      />
    </div>
    <h1
      className="project-name"
      data-jsx={679893488}
      title="sample"
    >
      sample
    </h1>
    <hr
      data-jsx={679893488}
    />
    <div
      className="box"
      data-jsx={679893488}
    >
      <img
        className="user-icon"
        data-jsx={679893488}
        src="https://avatars0.githubusercontent.com/u/12455918?v=4"
      />
      <div
        className="username"
        data-jsx={679893488}
      />
      <div
        className="date"
        data-jsx={679893488}
      >
        
      </div>
      <div
        className="description"
        data-jsx={679893488}
      >
        this is sample project
      </div>
    </div>
  </div>
</div>
`;
